src/Interpreters/ExpressionAnalyzer.cpp

```diff
@@ -330,46 +333,107 @@ void ExpressionAnalyzer::analyzeAggregation(ActionsDAGPtr & temp_actions)
                 if (getContext()->getSettingsRef().enable_positional_arguments)
                     replaceForPositionalArguments(group_asts[i], select_query, ASTSelectQuery::Expression::GROUP_BY);
 
-                getRootActionsNoMakeSet(group_asts[i], temp_actions, false);
-
-                const auto & column_name = group_asts[i]->getColumnName();
+                if (select_query->group_by_with_grouping_sets)
+                {
+                    ASTs group_elements_ast;
+                    const ASTExpressionList * group_ast_element = group_asts[i]->as<const ASTExpressionList>();
+                    if (!group_ast_element)
+                        throw Exception("Grouping Sets element " + group_asts[i]->getColumnName() + " should be an expression type", ErrorCodes::UNKNOWN_IDENTIFIER);
```

Comment:
Here it already should be logical error (we have incorrectly parsed AST?)
(also fmt-like syntax is better: `throw Exception(ErrorCodes::UNKNOWN_IDENTIFIER, "Grouping Sets element {} should be an expression type", group_asts[i]->getColumnName())`)